chill-hadoop/src/main/java/com/twitter/chill/hadoop/Varint.java

```diff
@@ -0,0 +1,168 @@
+        writeUnsignedVarInt((value << 1) ^ (value >> 31), out);
+    }
+
+    /**
+     * @see #writeUnsignedVarLong(long, java.io.DataOutputStream)
+     */
+    public static void writeUnsignedVarInt(int value, DataOutputStream out) throws IOException {
+        while ((value & 0xFFFFFF80) != 0L) {
+            out.writeByte((value & 0x7F) | 0x80);
+            value >>>= 7;
+        }
+        out.writeByte(value & 0x7F);
+    }
+
+    /**
+     * @param in to read bytes from
+     * @return decode value
+     * @throws java.io.IOException if {@link java.io.DataInput} throws {@link java.io.IOException}
+     * @throws IllegalArgumentException if variable-length value does not terminate
+     *  after 9 bytes have been read
+     * @see #writeSignedVarLong(long, java.io.DataOutputStream)
+     */
+    public static long readSignedVarLong(DataInputStream in) throws IOException {
+        long raw = readUnsignedVarLong(in);
+        // This undoes the trick in writeSignedVarLong()
+        long temp = (((raw << 63) >> 63) ^ raw) >> 1;
+        // This extra step lets us deal with the largest signed values by treating
+        // negative results from read unsigned methods as like unsigned values
+        // Must re-flip the top bit if the original read value had it set.
+        return temp ^ (raw & (1L << 63));
+    }
+
+    /**
+     * @param in to read bytes from
+     * @return decode value
+     * @throws java.io.IOException if {@link java.io.DataInput} throws {@link java.io.IOException}
+     * @throws IllegalArgumentException if variable-length value does not terminate
+     *  after 9 bytes have been read
+     * @see #writeUnsignedVarLong(long, java.io.DataOutputStream)
+     */
+    public static long readUnsignedVarLong(DataInputStream in) throws IOException {
+        long value = 0L;
+        int i = 0;
+        long b;
+        while (((b = in.readByte()) & 0x80L) != 0) {
+            value |= (b & 0x7F) << i;
+            i += 7;
+        }
+        return value | (b << i);
+    }
+
+    /**
+     * @throws IllegalArgumentException if variable-length value does not terminate
+     *  after 5 bytes have been read
+     * @throws java.io.IOException if {@link java.io.DataInput} throws {@link java.io.IOException}
+     * @see #readSignedVarLong(java.io.DataInputStream)
+     */
+    public static int readSignedVarInt(DataInputStream in) throws IOException {
+        int raw = readUnsignedVarInt(in);
+        // This undoes the trick in writeSignedVarInt()
+        int temp = (((raw << 31) >> 31) ^ raw) >> 1;
+        // This extra step lets us deal with the largest signed values by treating
+        // negative results from read unsigned methods as like unsigned values.
+        // Must re-flip the top bit if the original read value had it set.
+        return temp ^ (raw & (1 << 31));
+    }
+
+    /**
+     * @throws IllegalArgumentException if variable-length value does not terminate
```

Comment:
Looks like one(this code isn't mine), add test, or kill comment? 

```
        while (((b = in.readByte()) & 0x80) != 0 && i < 42) {
            value |= (b & 0x7F) << i;
            i += 7;
        }
        if(i == 42) { // Over read!
            throw IllegalArgumentException("Read more than 5 bytes of data, must be invalid Var int")
        }
```
Looks like it would fix this one